### PR TITLE
Support for the new 0.61.0 API

### PR DESCRIPTION
**Pull request by RebliNk17**
Added support for the new 0.61.0 api.

Added a Settings listener.
The settings listener should be add as the other listeners in the api.

Need to update protos.
For some reason that didn't work for me.

### CODE REVIEW

**Review by RebliNk17 — COMMENTED** [review]
ignore that

File: library/src/main/java/com/pokegoapi/api/settings/Settings.java [hunk 1]
```diff
@@ -123,6 +126,30 @@ public void updateSettings(DownloadSettingsResponse response) {
 		if (response.getSettings().hasGpsSettings()) {
 			gpsSettings.update(response.getSettings().getGpsSettings());
 		}
+
+		// check if new api has been forced by Niantic
+		if(!response.getSettings().getMinimumClientVersion().isEmpty()) {
+			int[] currentVer = api.getHashProvider().getCurrentAPIVersion();
+			String[] minimumVer = response.getSettings().getMinimumClientVersion().split("\\.");
+			int[] splitMinimumVer = new int[currentVer.length];
```
Comment (Gegy) [inline library/src/main/java/com/pokegoapi/api/settings/Settings.java:134]: `minimumVer` should be `splitMinimumVer`. What is currently `spltMinimumVer` should not use the length of `currentVer`, but the length of what is currently `minimumVer`. You should probably also handle the case of `currentVer` and `minimumVer` having different lengths.

File: library/src/resources/.gitignore [hunk 2]
```diff
@@ -1,2 +0,0 @@
-compile.bat
-protoc.exe
```
Comment (Gegy) [inline library/src/resources/.gitignore:2]: What's this?

File: library/src/main/java/com/pokegoapi/api/settings/Settings.java [hunk 3]
```diff
@@ -123,6 +126,30 @@ public void updateSettings(DownloadSettingsResponse response) {
 		if (response.getSettings().hasGpsSettings()) {
 			gpsSettings.update(response.getSettings().getGpsSettings());
 		}
+
+		// check if new api has been forced by Niantic
+		if(!response.getSettings().getMinimumClientVersion().isEmpty()) {
```
Comment (jabbink) [inline library/src/main/java/com/pokegoapi/api/settings/Settings.java:131]: [ant:checkstyle] [WARN] /home/jasper/Documents/PokemonGo/PokeGOAPI-Java/library/src/main/java/com/pokegoapi/api/settings/Settings.java:131:7: WhitespaceAround: 'if' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]

File: library/src/main/java/com/pokegoapi/api/listener/SettingsListener.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+public interface SettingsListener extends Listener {
+
+	/**
+	 * Called when a new API is forced
+	 *
+	 * @param go the current api
+	 * @param currentVersion the current supported api version
+	 * @param forcedVersion the new forced api.
+	 */
+	 void onNewVersionForced(PokemonGo go, String currentVersion, String forcedVersion);
```
Comment (jabbink) [inline library/src/main/java/com/pokegoapi/api/listener/SettingsListener.java:17]: [ant:checkstyle] [WARN] /home/jasper/Documents/PokemonGo/PokeGOAPI-Java/library/src/main/java/com/pokegoapi/api/listener/SettingsListener.java:17: Indent must use tab characters [RegexpSinglelineJava]

Comment (jabbink) [reply library/src/main/java/com/pokegoapi/api/listener/SettingsListener.java:17]: [ant:checkstyle] [WARN] /home/jasper/Documents/PokemonGo/PokeGOAPI-Java/library/src/main/java/com/pokegoapi/api/listener/SettingsListener.java:17: 'method def modifier' have incorrect indentation level 3, expected level should be 2. [Indentation]

File: library/src/main/java/com/pokegoapi/api/listener/SettingsListener.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package com.pokegoapi.api.listener;
+
+import com.pokegoapi.api.PokemonGo;
+
+/**
+ * Created by RebliNk17 on 09/04/2017.
+ */
+public interface SettingsListener extends Listener {
+
+	/**
```
Comment (jabbink) [inline library/src/main/java/com/pokegoapi/api/listener/SettingsListener.java:10]: [ant:checkstyle] [WARN] /home/jasper/Documents/PokemonGo/PokeGOAPI-Java/library/src/main/java/com/pokegoapi/api/listener/SettingsListener.java:10: Block comment has incorrect indentation level 1, expected is 2, indentation should be the same level as line 17. [CommentsIndentation]

**Review by jabbink — CHANGES_REQUESTED** [review]
Check the style conventions

**Pull request closed without merging**